nifi-nar-bundles/nifi-spark-bundle/nifi-livy-processors/src/main/java/org/apache/nifi/processors/livy/ExecuteSparkInteractive.java

```diff
@@ -178,48 +247,188 @@ public void onTrigger(ProcessContext context, final ProcessSession session) thro
+                        session.transfer(flowFile, REL_FAILURE);
+                        return;
+                    }
+                }
+
+                log.debug(" ====> jsonResponse: " + jsonResponse);
+
                 try {
-                    final JSONObject output = result.getJSONObject("data");
-                    flowFile = session.write(flowFile, out -> out.write(output.toString().getBytes()));
-                    flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), LivySessionService.APPLICATION_JSON);
-                    session.transfer(flowFile, REL_SUCCESS);
-                } catch (JSONException je) {
-                    // The result doesn't contain the data, just send the output object as the flow file content to failure (after penalizing)
-                    log.error("Spark Session returned an error, sending the output JSON object as the flow file content to failure (after penalizing)");
-                    flowFile = session.write(flowFile, out -> out.write(result.toString().getBytes()));
+
+                    final JSONObject jsonResponseObj = new JSONObject(jsonResponse);
+
+                    Map<String, String> headers = new HashMap<>();
+                    headers.put("Content-Type", LivySessionService.APPLICATION_JSON);
+                    headers.put("X-Requested-By", LivySessionService.USER);
+                    headers.put("Accept", "application/json");
+
+                    JSONObject jobInfo = readJSONObjectFromUrl(jsonResponseObj.getString("url"), livySessionService, headers);
+
+                    flowFile = session.write(flowFile, out -> out.write(jobInfo.toString().getBytes()));
                     flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), LivySessionService.APPLICATION_JSON);
-                    flowFile = session.penalize(flowFile);
+
+                    Thread.sleep(statusCheckInterval);
+
+                    String state  = jobInfo.getString("state");
+                    log.debug(" ====> jsonResponseObj State: " + state);
+
+                    switch (state) {
+                        case "success":
+                            log.debug(" ====> success State: " + state);
+                            session.transfer(flowFile, REL_SUCCESS);
+                            break;
+                        case "dead":
+                            log.debug(" ====> dead State: " + state);
+                            session.transfer(flowFile, REL_FAILURE);
+                            break;
+                        default:
+                            log.debug(" ====> default State: " + state);
+                            session.transfer(flowFile, REL_WAIT);
+                            break;
+                    }
+
+                } catch (JSONException | InterruptedException e) {
+
+                    //Incoming flow file is not an JSON file hence consider it to be an triggering point
```

Comment:
Cosmetic change: Multiple empty lines were left. IMHO, one empty line should be enough for better readability.